be1-go/channel/election/mod_test.go

```diff
@@ -476,13 +476,10 @@ func Test_Sending_Election_Key(t *testing.T) {
 
 	electionKeyMsg := catchupAnswer[1]
 
-	data := messagedata.ElectionKey{}.NewEmpty()
+	var dataKey messagedata.ElectionKey
 
-	err := electionKeyMsg.UnmarshalData(data)
-	require.NoError(t, err)
-
-	dataKey, ok := data.(*messagedata.ElectionKey)
-	require.True(t, ok)
```

Comment:
why dropping the check on the data ?

Reply:
The check on the data is simply a cast, with the check itself verifying if the cast worked or not. It is in my opinion unnecssary as dataKey already has the correct type.
For the failure in itself, it could be this, even if the error message didn't indicate it (it was only "unexpected end of JSON message"), and if so it is fixed now.

Reply:
Thanks !